api/src/main/java/org/openmrs/validator/PersonValidator.java

```diff
@@ -121,6 +122,19 @@ private void validateBirthDate(Errors errors, Date birthDate) {
 		rejectIfFutureDate(errors, birthDate, "birthdate");
 		rejectDateIfBefore120YearsAgo(errors, birthDate, "birthdate");
 	}
+
+	/**
+	 * Checks if the death date is in accordance with dead status.
+	 *
+	 * @param errors Stores information about errors encountered during validation
+	 * @param deathDate the deathdate to validate
+	 * @param dead is the dead status
+	 */
+	private void validateDeathDateDead(Errors errors, Date deathDate, boolean dead, boolean deathDateEstimated) {
+		if(deathDate != null && dead == false && deathDateEstimated == false) {
+			errors.rejectValue("deathDate", "person.dead.false");
```

Comment:
i was to specify that that the attribute dead of the person class was false and due to this the death date can not be set unless we set dead to true.